pkgs/development/libraries/science/math/osqp/default.nix

```
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, cmake, lib }:
+
+stdenv.mkDerivation {
+  pname = "osqp";
+  version = "2020-07-17";
+
+  src = fetchFromGitHub {
+    owner  = "oxfordcontrol";
+    repo   = "osqp";
+    rev    = "38429297b8c48c3189e2ca7eac97e7e821275bab";
+    sha256 = "1fn5pb6i4apdwmwifkyva8l7lw04vsqg5p299782fcy12v1ldcdz";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ cmake ];
+
+  meta = {
+    homepage = "https://osqp.org";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.taktoa ];
+    platforms = lib.platforms.linux;
```

Why restrict platforms? Is there anything keeping it from being built on e.g. Mac? 
```suggestion
    platforms = platforms.all;
```

---

I just haven't tested it on other platforms